silk/tests/test_unit.py

```diff
@@ -197,3 +197,18 @@ def test_metrics_to_collect(instance, enable_rw, enable_bs, extra_metrics_to_col
     expected_metrics_to_collect = deepcopy(METRICS)
     expected_metrics_to_collect.update(extra_metrics_to_collect)
     assert sorted(check.metrics_to_collect.keys()) == sorted(expected_metrics_to_collect.keys())
+
+
+def test_unreachable_endpoint(dd_run_check, aggregator):
+    invalid_instance = {'host_address': 'http://{}:81'.format(HOST)}
+    check = SilkCheck('silk', {}, [invalid_instance])
+
+    with pytest.raises(Exception):
+        dd_run_check(check)
+    aggregator.assert_service_check('silk.can_connect', SilkCheck.CRITICAL)
+
+
+def test_incorrect_config(dd_run_check):
+    invalid_instance = {'host_addres': 'localhost'}  # misspelled required parameter
+    with pytest.raises(ConfigurationError):
+        SilkCheck('silk', {}, [invalid_instance])
```

Comment:
These two tests do not interact with an external system and do not need the integration environment to be up. You could argue that they're not purely unit test since they're exercising more than one method/class/module but for our purposes having the environment up or not is a good enough distinction.